plugins/repositories-plugin/src/main/java/org/pentaho/di/ui/repo/RepositoryConnectController.java

```diff
@@ -220,11 +220,13 @@ public void addDatabase( DatabaseMeta databaseMeta ) {
   }
 
   public boolean setDefaultRepository( String name ) {
-    for ( int i = 0; i < repositoriesMeta.nrRepositories(); i++ ) {
-      repositoriesMeta.getRepository( i ).setDefault( false );
-    }
     RepositoryMeta repositoryMeta = repositoriesMeta.findRepository( name );
-    repositoryMeta.setDefault( true );
+    if( repositoryMeta != null ) {
```

Comment:
@ddiroma Just a small checkstyle issue with a missing space between the if and (.